src/Avalonia.Controls/Templates/TemplateExtensions.cs

```diff
@@ -4,14 +4,47 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using Avalonia.Controls;
 using Avalonia.Styling;
 using Avalonia.VisualTree;
 
 namespace Avalonia.Controls.Templates
 {
     public static class TemplateExtensions
     {
+        /// <summary>
+        /// Gets a named control from a templated control's template children.
+        /// </summary>
+        /// <param name="template">The control template.</param>
+        /// <param name="name">The name of the control.</param>
+        /// <param name="templatedParent">The templated parent control.</param>
+        /// <returns>An <see cref="IControl"/> or null if the control was not found.</returns>
+        public static IControl FindName(this IControlTemplate template, string name, IControl templatedParent)
+        {
+            Contract.Requires<ArgumentNullException>(template != null);
+            Contract.Requires<ArgumentNullException>(name != null);
+            Contract.Requires<ArgumentNullException>(templatedParent != null);
+
+            return ((IControl)templatedParent.GetVisualChildren().FirstOrDefault())?.FindControl<Canvas>(name);
+        }
+
+        /// <summary>
+        /// Gets a named control from a templated control's template children.
+        /// </summary>
+        /// <typeparam name="T">The type of the template child.</typeparam>
+        /// <param name="template">The control template.</param>
+        /// <param name="name">The name of the control.</param>
+        /// <param name="templatedParent">The templated parent control.</param>
+        /// <returns>An <see cref="IControl"/> or null if the control was not found.</returns>
+        public static T FindName<T>(this IControlTemplate template, string name, IControl templatedParent)
+            where T : class, IControl
+        {
+            Contract.Requires<ArgumentNullException>(template != null);
+            Contract.Requires<ArgumentNullException>(name != null);
+            Contract.Requires<ArgumentNullException>(templatedParent != null);
+
+            return template.FindName(name, templatedParent) as T;
```

Comment:
I do not think use `as` keyword is a good way. Some of my friends may change the control type and I will get the null exception and I can not fast to find the control type be changed.

Could you add some code to check the control name is exist and the control type is right?

Reply:
That's not the desired behavior most of the time. Missing PART contracts should not break the control.

Reply:
The common convention for `FindXXX`/`GetXXX` is that `Find` returns null if not found and `Get` throws an exception.

Reply:
I can't think of a reason where it would matter, but if this _does_ matter, than you could just do `FindControl("name")` or `FindControl<IControl>("name")` and do the cast yourself.